server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -1363,6 +1364,20 @@ public GatedCloseable<IndexCommit> acquireLastIndexCommit(boolean flushFirst) th
         }
     }
 
+    public Optional<NRTReplicationEngine> getReplicationEngine() {
+        if (getEngine() instanceof NRTReplicationEngine) {
```

Comment:
Why does `IndexShard` specifically need to know about `NRTReplicationEngine ` can we simply return an `Engine` reference instead?

Reply:
@Poojita-Raj we basically expose whole engine here which we were trying to avoid in the first place https://github.com/opensearch-project/OpenSearch/pull/3533#discussion_r893927672

Reply:
@reta - Ah I see, I misunderstood. I've made the change to add the method updateSegments to Engine that by default throws an UnsupportedException, and which NRTReplicationEngine overrides to implement.

Reply:
Thanks @Poojita-Raj , the interface (like`ReplicationEngine`) would solve this ambiguity: if the engine does not implement it - no instance is returned.